docs/introduction/how-to-work-with-money.md

```diff
@@ -0,0 +1,388 @@
+
+All `price*` Twig filters use this format.
+They usually differ only in the currency and locale they use.
+
+### priceText
+
+Filter `priceText` formats the amount of money in a localized manner, similarly to the `price` filter.
+The only difference is that it outputs the text *"Free"* when zero amount of money is provided.
+
+### priceTextWithCurrencyByCurrencyIdAndLocale
+
+Filter `priceTextWithCurrencyByCurrencyIdAndLocale` can be used to format the amount of money in any currency and locale.
+It outputs the text *"Free"* when zero amount of money is provided.
+
+The *currency ID* (`int`) and the *locale* (`string`) must be provided as parameters.
+
+### priceWithCurrency
+
+Filter `priceWithCurrency` can be used to format the amount of money in any currency.
+It uses the current locale (the locale of current domain or administration) to format it.
+
+The *currency* (`Currency` entity) must be provided as a parameter.
+
+### priceWithCurrencyAdmin
+
+It works similarly to `priceWithCurrency`, but it uses the default administration currency.
+
+It has no parameters.
+
+### priceWithCurrencyByCurrencyId
+
+It works similarly to `priceWithCurrency`, but it uses currency ID instead of the `Currency` entity.
+
+The *currency ID* (`int`) must be provided as a parameter.
+
+### priceWithCurrencyByDomainId
+
+It works similarly to `priceWithCurrency`, but it uses the basic frontend currency of the provided domain.
+
+The *domain ID* (`int`) must be provided as a parameter.
+
+## Money in Javascript
+
+`Money` implements the `JsonSerializable` interface, which means it can be serialized into JSON format with `json_encode` into an object with the following structure:
+
+```json
+{
+  "amount": "1234.5670"
+}
+```
+
+This format can be readily used in Javascript when JSON is rendered in Twig:
+
+```twig
+{# the "money" variable contains Money::create('1234.5670') #}
+
+<div id="my-money" data-money="{{ money|json_encode }}"/>
+```
+
+```js
+(function ($) {
+
+    $(document).ready(function () {
+        var money = $('#my-money').data('money');
+
+        // Displays "1234.5670"
+        alert(money.amount);
+    });
+
+})(jQuery);
+```
+
+See the native Javascript method [`.toFixed()`](https://www.w3schools.com/jsref/jsref_tofixed.asp) and the function [`parseFloat()`](https://www.w3schools.com/jsref/jsref_parsefloat.asp) to see how to convert between decimal strings and numbers.
+
+## Money in Doctrine
+
+In Shopsys Framework, there is a custom Doctrine type `money` which can be used in similar fashion as `decimal` type.
+The entity property value will be automatically hydrated to an instance of `Money` if it's configured to use the type:
+
+```php
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Table(name="my_entities")
+ * @ORM\Entity
+ */
+class MyEntity
+{
+    /**
+     * @var \Shopsys\FrameworkBundle\Component\Money\Money|null
+     *
+     * @ORM\Column(type="money", precision=20, scale=6, nullable=true)
```

Comment:
The example is quite long and the most important part `type="money"` looses in the context. Please remove class, uses and keep here only the field, so it will be highlighted.
I know it won't be 100% correct, but I prefer simplicity over correctness ;-)

Reply:
I've removed the `@ORM\Table` in 504fc6b94 which wasn't required. I think it's important to note the alias `use Doctrine\ORM\Mapping as ORM;` even though it's common in our code and the fact that I'm talking about entities.

Reply:
see the difference
```php
use Doctrine\ORM\Mapping as ORM;

/**
 * @ORM\Entity
 */
class MyEntity
{
    /**
     * @var \Shopsys\FrameworkBundle\Component\Money\Money|null
     *
     * @ORM\Column(type="money", precision=20, scale=6, nullable=true)
     */
    protected $price;

    // ...
}
```

```php
// class MyEntity

    /**
     * @var \Shopsys\FrameworkBundle\Component\Money\Money|null
     *
     * @ORM\Column(type="money", precision=20, scale=6, nullable=true)
     */
    protected $price;
```

shorter (half the length), maybe not so correct, but highlighting the most important part

Reply:
I think the verbosity is not such a big deal here and I'd like to keep it as it is. It better communicates the concept of the entity class and it explains the `@ORM\Column` annotation.